authors/apps/authentication/views.py

```diff
@@ -446,3 +449,47 @@ def get(self, request):
         queryset = self.get_queryset()
         serializer = ProfilesSerializer(queryset, many=True)
         return Response({"Profiles": serializer.data})
+
+
+class SubscribeAPIView(APIView):
+    """Allows user to subscribe or unsubscribe from notifications"""
+
+    permission_classes = (AllowAny,)
+    serializer_class = SubscriptionSerializer
+
+    def put(self, request):
+        user = User.objects.get(email=request.user.email)
+        if not user.get_notifications:
+            user.get_notifications = True
+            user.save()
+            return Response({
+                "message": "You have successfully subscribed"
+            }, status=status.HTTP_200_OK)
+
+        else:
+            return Response({
+                "Message": "You are already subscribed"
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+
+class UnsubscribeAPIView(APIView):
+    "Allows user to unsubscribe to notifications"
+
+    permission_classes = (AllowAny,)
+
+    serializer_class = SubscriptionSerializer
+
+    def get(self, request, uuid):
```

Comment:
Using GET for unsubscribing is a bit confusing. I suggest you use put

Reply:
done